new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java

```diff
@@ -0,0 +1,387 @@
+   * Install 2.5.0 release Operator from GitHub chart repository and create a domain.
+   * Delete Operator and install latest Operator and verify CRD version is updated
+   * and the domain can be managed by scaling the cluster using operator REST api.
+   */
+  @Test
+  @DisplayName("Upgrade Operator from 2.5.0 to latest")
+  @MustNotRunInParallel
+  public void testOperatorUpgradeFrom2_5_0(@Namespaces(3) List<String> namespaces) {
+    this.namespaces = namespaces;
+    upgradeOperator("2.5.0", false);
+  }
+
+  /**
+   * Operator upgrade from 2.6.0 to latest.
+   * Install 2.6.0 Operator from GitHub chart repository and create a domain.
+   * Delete Operator and install latest Operator and verify CRD version is updated
+   * and the domain can be managed by scaling the cluster using operator REST api.
+   */
+  @Test
+  @DisplayName("Upgrade Operator from 2.6.0 to latest")
+  @MustNotRunInParallel
+  public void testOperatorUpgradeFrom2_6_0(@Namespaces(3) List<String> namespaces) {
+    this.namespaces = namespaces;
+    upgradeOperator("2.6.0", false);
+  }
+
+  /**
+   * Operator upgrade from 3.0.0 to latest.
+   * Install 3.0.0 Operator from GitHub chart repository and create a domain.
+   * Upgrade operator with latest Operator image and verify CRD version and image are updated
+   * and the domain can be managed by scaling the cluster using operator REST api.
+   */
+  @Test
+  @DisplayName("Upgrade Operator from 3.0.0 to latest")
+  @MustNotRunInParallel
+  public void testOperatorUpgradeFrom3_0_0(@Namespaces(3) List<String> namespaces) {
+    this.namespaces = namespaces;
+    upgradeOperator("3.0.0", true);
+  }
+
+  /**
+   * Cleanup Kubernetes artifacts in the namespaces used by the test and
+   * delete CRD.
+   */
+  @AfterEach
+  public void tearDown() {
+    if (System.getenv("SKIP_CLEANUP") == null) {
+      CleanupUtil.cleanup(namespaces);
+      new Command()
+          .withParams(new CommandParams()
+              .command("kubectl delete crd domains.weblogic.oracle --ignore-not-found"))
+          .execute();
+    }
+  }
+
+  private void upgradeOperator(String operatorVersion, boolean useHelmUpgrade) {
+    logger.info("Assign a unique namespace for operator {0}", operatorVersion);
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    final String opNamespace1 = namespaces.get(0);
+    logger.info("Assign a unique namespace for latest operator");
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    final String opNamespace2 = namespaces.get(1);
+    logger.info("Assign a unique namespace for domain");
+    assertNotNull(namespaces.get(2), "Namespace is null");
+    String domainNamespace = namespaces.get(2);
+    latestOperatorImageName = getOperatorImageName();
+
+    // delete existing CRD
+    new Command()
+        .withParams(new CommandParams()
+            .command("kubectl delete crd domains.weblogic.oracle --ignore-not-found"))
+        .execute();
```

Comment:
what happens when the test fails after this stage? The subsequent tests fails without a CRD?

Reply:
The next step in the test is to install Operator which creates CRD. If the test fails after deleting the CRD and before installing the operator, for the next upgrade test the CRD will be created when the operator is installed. subsequent tests will not fail.

Reply:
My concern is if any of the upgrade test(may be the last upgradetest) fails and leaves the env with a missing CRD it is going to affect the other tests in other IT classes right?

Reply:
The idea is upgrade tests should delete the CRD(as it uses old operator releases) after the testing is done(success or failed). All the IT classes install the operator first, this is when CRD is created hence there won't be a situation where the tests will be run without CRD.